aws-connect-user/src/main/java/software/amazon/connect/user/UpdateHandler.java

```diff
@@ -85,6 +89,18 @@
         }
     }
 
+    private boolean isUserIdentityInfoUpdated(final UserIdentityInfo desiredUserIdentityInfo, final UserIdentityInfo previousUserIdentityInfo) {
+        if (desiredUserIdentityInfo == null && previousUserIdentityInfo == null) {
+            return false;
+        }
+        if (desiredUserIdentityInfo == null || previousUserIdentityInfo == null) {
+            return true;
+        }
+        return !desiredUserIdentityInfo.getFirstName().equals(previousUserIdentityInfo.getFirstName()) ||
+                !desiredUserIdentityInfo.getLastName().equals(previousUserIdentityInfo.getLastName()) ||
+                !desiredUserIdentityInfo.getEmail().equals(previousUserIdentityInfo.getEmail());
+    }
+
     private static UpdateUserIdentityInfoRequest translateToUpdateUserIdentityInfoRequest(final ResourceModel model) {
```

Comment:
Isn't deep equals implemented on `UserIdentityInfo` which we can use versus writing our own logic?

Reply:
This object is auto-generated and does not have equals methods